### PR TITLE
Further documentation for derivatives by array

### PR DESCRIPTION
**Pull request by Upabjojr**
LaTeX definitions and example usages.

### CODE REVIEW

File: sympy/tensor/array/__init__.py [hunk 1]
```diff
@@ -156,6 +156,48 @@
...
+respect to arrays, provided that all elements in the that array are symbols or
+expressions suitable for derivations.
+
+The definition of a derivative by an array is as follows: given the array
+`A_{i_1, \ldots, i_N}` and the array `X_{j_1, \ldots, j_M}`
+the derivative of arrays will return a new array `B` defined by
+
+`B_{j_1,\ldots,j_M,i_1,\ldots,i_N} := \\frac{\partial A_{i_1,\ldots,i_N}}{\partial X_{j_1,\ldots,j_M}}`
+
+The function ``derive_by_array`` performs such derivative:
```
Comment (jksuom) [inline sympy/tensor/array/__init__.py]: Maybe: such an operation  (?)

**Comment by Upabjojr** [comment]
@jksuom your comment has been fixed. I also turned strings into raw strings in order to escape all backslashes in LaTeX expressions. Hopefully fixes https://github.com/sympy/sympy/issues/10599

**Comment by jksuom** [comment]
Looks good to me. Is there any reason not to merge before #10599 is cleared up?

**Comment by Upabjojr** [comment]
> Is there any reason not to merge before #10599 is cleared up?

I hope the problem in #10599 is the lack of raw string, so if we merge this PR, that issue could be solved (unless the problem is something else).

On my computer the documentation builds correctly, so I'm not able to determine the problem with the server documentation.

**Comment by jksuom** [comment]
Thanks, that is plausible. Merging.

**Pull request merged**